dotnet/samples/04.e.twentyQuestions/Program.cs

```diff
@@ -129,14 +136,21 @@
         moderator: moderator,
         prompt: "Chat",
         history: aiHistoryOptions);
-    ApplicationOptions<GameState, GameStateManager> applicationOptions = new()
+    var applicationBuilder = new ApplicationBuilder<GameState, GameStateManager>()
+    .WithAIOptions(aiOptions)
+    .WithLoggerFactory(loggerFactory)
+    .WithTurnStateManager(new GameStateManager());
+
+    // Set storage options
+    IStorage? storage = sp.GetService<IStorage>();
+    if (storage != null)
     {
-        TurnStateManager = new GameStateManager(),
-        Storage = sp.GetService<IStorage>(),
-        AI = aiOptions,
-        LoggerFactory = loggerFactory,
-    };
-    Application<GameState, GameStateManager> app = new(applicationOptions);
+        applicationBuilder.WithStorage(storage);
+    }
+
+    // Create Application
+    Application<GameState, GameStateManager> app = applicationBuilder.Build();
```

Comment:
Instead of declaring the applicationBuilder and then doing app = applicationBuilder.build() you can do app = ApplicationBuilder.withX().withY().Build().

I find that to be more concise. Thoughts?